File: backend/src/main/java/com/gxs/bank/model/FixedDeposit.java
package com.gxs.bank.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.*;
import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.UUID;

@Entity
@Table(name = "fixed_deposits")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class FixedDeposit {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    @JsonIgnore
    private User user;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "source_account_id", nullable = false)
    @JsonIgnore
    private SavingsAccount sourceAccount;

    @Column(name = "principal_amount", nullable = false, precision = 15, scale = 2)
    private BigDecimal principalAmount;

    @Column(name = "interest_rate", nullable = false, precision = 5, scale = 2)
    private BigDecimal interestRate;

    @Column(name = "tenure_months", nullable = false)
    private Integer tenureMonths;

    @Column(name = "maturity_date", nullable = false)
    private LocalDate maturityDate;

    @Column(name = "maturity_amount", nullable = false, precision = 15, scale = 2)
    private BigDecimal maturityAmount;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private Status status;

    @Column(name = "auto_renew")
    private Boolean autoRenew;

    @Column(name = "fd_number", unique = true)
    private String fdNumber;

    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        if (status == null) status = Status.ACTIVE;
        if (autoRenew == null) autoRenew = false;
        if (fdNumber == null) fdNumber = "FD" + String.format("%06d", System.nanoTime() % 1000000);
    }

    public enum Status { ACTIVE, MATURED, BROKEN, RENEWED }
}

File: backend-python/com/gxs/bank/model/FixedDeposit.py
from datetime import date, datetime, timedelta
from decimal import Decimal
from enum import Enum
import random
import uuid

from sqlalchemy import Boolean, Column, Date, DateTime, Enum as SQLEnum, ForeignKey, Integer, Numeric, String
from sqlalchemy.orm import relationship

from com.gxs.bank.config.database import Base


class Status(str, Enum):
    ACTIVE = "ACTIVE"
    MATURED = "MATURED"
    BROKEN = "BROKEN"
    RENEWED = "RENEWED"


def _default_fd_number() -> str:
    return f"FD{random.randint(0, 999999):06d}"


def _default_maturity_date() -> date:
    return date.today() + timedelta(days=30)


class FixedDeposit(Base):
    __tablename__ = "fixed_deposits"

    id = Column(String(36), primary_key=True, default=lambda: str(uuid.uuid4()))
    userId = Column("user_id", String(36), ForeignKey("users.id"), nullable=False)
    user = relationship("User", lazy="select")

    sourceAccountId = Column("source_account_id", String(36), ForeignKey("savings_accounts.id"), nullable=False)
    sourceAccount = relationship("SavingsAccount", lazy="select")

    principalAmount = Column("principal_amount", Numeric(15, 2), nullable=False)
    interestRate = Column("interest_rate", Numeric(5, 2), nullable=False)
    tenureMonths = Column("tenure_months", Integer, nullable=False)
    maturityDate = Column("maturity_date", Date, nullable=False, default=_default_maturity_date)
    maturityAmount = Column("maturity_amount", Numeric(15, 2), nullable=False, default=Decimal("0.00"))
    status = Column(SQLEnum(Status), nullable=False, default=Status.ACTIVE)
    autoRenew = Column("auto_renew", Boolean, nullable=False, default=False)
    fdNumber = Column("fd_number", String(255), unique=True, nullable=False, default=_default_fd_number)
    createdAt = Column("created_at", DateTime, nullable=False, default=datetime.utcnow)


FixedDeposit.Status = Status
